gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -210,7 +218,10 @@ public static ClientContext create(StubSettings settings) throws IOException {
         .setClock(clock)
         .setDefaultCallContext(defaultCallContext)
         .setEndpoint(settings.getEndpoint())
-        .setQuotaProjectId(settings.getQuotaProjectId())
+        .setQuotaProjectId(
+            settings.getQuotaProjectId() == null && quotaProjectID != null
+                ? quotaProjectID
+                : settings.getQuotaProjectId())
```

Comment:
I'm not convinced that we need this -- the internal quotaProjectId here is only used in the constructor when converting back to StubSettings. Since the headers, internalHeaders, and credentials will still be set, if the resulting StubSettings is converted back to ClientContext, the same quotaProjectId resolution will happen.

Reply:
As I understand from our chat, `Settings would only care about the explicit value set from setQuotaProjectId
ClientContext would care about all 4 sources of that value`. 

1. `Settings would only care about the explicit value set from setQuotaProjectId`, so I removed get `QuotaProjectId` from [setHeaderProvider](https://github.com/googleapis/gax-java/pull/1166/files/eb7da4f5e990cd94ad93a1b0c4c67c506ef06571#diff-cd638eb9960c43bba008c9fc6ffe1e1fL290), [setInternalHeaderProvider](https://github.com/googleapis/gax-java/pull/1166/files/eb7da4f5e990cd94ad93a1b0c4c67c506ef06571#diff-cd638eb9960c43bba008c9fc6ffe1e1fL307), and [SetCredentialProvider](https://github.com/googleapis/gax-java/pull/1166/files/eb7da4f5e990cd94ad93a1b0c4c67c506ef06571#diff-cd638eb9960c43bba008c9fc6ffe1e1fL269).

2. `ClientContext would care about all 4 sources of that value`: (summary follow, if quotaProjectId not being set from setting, we update the value from clientcontext if it has been set from header, internalheader or credentials)
Details:
when create ClientContext,
first, we check if quota_project_id has been set from setting, if yes, we just use this value, update header with this value and wrap credentials hide quotaProjectId
if not, then, we look at if header and internalHeader contain quota-project-id key, if yes, we update local quota project id, and wrap credentials hide quotaProjectId. 
final, we look at if credential has quotaProjectId key, if yes we update local quota project id
At the end,  clientcontext.setQuotaProjectId with correct value.

Reply:
My comment is that the instance value set by clientcontext.setQuotaProjectId is not how the header is being set on the request and is only used when recreating a StubSettings instance from the ClientContext.

Reply:
It doesn't hurt to do this (except doing an extra conditional check above), but it doesn't seem necessary.

Reply:
I guess I know why it is necessary, because https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java#L199-L213

Maybe I over think this problem. Looks like we only need to remove credential from stubsetting. Your PR is already covered everything. 

I am agreeing on #1162